### PR TITLE
[JENKINS-32273] Jenkins CLI unification of error handling and returning

### PR DESCRIPTION
**Pull request by pjanouse**
https://issues.jenkins-ci.org/browse/JENKINS-32273

Extended exception handling and logging in CLICommand class.

This is the first step which shouldn't break anything existing (fully backward compatible).

Next (dependent) part will follow - to re-factorize CLI commands already implemented in CLI module to unify error handling and returning - this part can't be fully backward compatible and I anticipate a longer discussion about it - that's the reason I'm dividing this effort to two separate parts and PRs. 

I believe it is worth and better for the future to do such an unification sooner than later even it isn't fully backward compatible.

### CODE REVIEW

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 1]
```diff
@@ -251,6 +255,10 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
             stderr.println("Bad Credentials. Search the server log for "+id+" for more details.");
             return -1;
         } catch (Exception e) {
+            final String errorMsg = String.format("Unexpected exception occurred during performing %s command: %s",
```
Comment (daniel-beck) [inline core/src/main/java/hudson/cli/CLICommand.java]: Grammar: I think *while performing* would be better.

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 2]
```diff
@@ -251,6 +255,10 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
...
+                    getName(), e.getMessage());
+            stderr.println(errorMsg);
+            LOGGER.log(Level.WARNING, errorMsg);
```
Comment (daniel-beck) [inline core/src/main/java/hudson/cli/CLICommand.java]: Will this lead to duplication of the error message on the client?

Comment (pjanouse) [inline core/src/main/java/hudson/cli/CLICommand.java]: No, this part of code is executed on master, the stderr is sent to the client, but log stays on the server where the destination depends on the log configuration of Jenkins master - common cfg. - final target is server log or the console.

File: core/src/main/java/hudson/cli/CLICommand.java (hunk 1 above)
Comment (pjanouse) [inline core/src/main/java/hudson/cli/CLICommand.java]: :+1:

**Comment by pjanouse** [comment]
Msg fixed.

File: core/src/main/java/hudson/cli/CLICommand.java [hunk 3]
```diff
@@ -251,6 +255,10 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
             stderr.println("Bad Credentials. Search the server log for "+id+" for more details.");
             return -1;
         } catch (Exception e) {
+            final String errorMsg = String.format("Unexpected exception occurred while performing %s command: %s",
+                    getName(), e.getMessage());
```
Comment (olivergondza) [inline core/src/main/java/hudson/cli/CLICommand.java]: If I am not mistaken the exception message will be printed by `printStackTrace`. No need to do it twice.

**Comment by pjanouse** [comment]
Log messages of unexpected exception improved.

**Pull request merged**